clar2wasm/src/wasm_generator.rs

```diff
@@ -538,6 +549,23 @@ impl WasmGenerator {
 
         self.early_return_block_id = Some(block_id);
 
+        // If the same arg name is used multiple times, the interpreter throws an
+        // `Unchecked` error at runtime, so we do the same here
+        if let Some(arg_name) = reused_arg {
+            let (arg_name_offset, arg_name_len) =
+                self.add_clarity_string_literal(&CharType::ASCII(ASCIIData {
+                    data: arg_name.as_bytes().to_vec(),
+                }))?;
+
+            block
+                .i32_const(arg_name_offset as i32)
+                .global_set(get_global(&self.module, "runtime-error-arg-offset")?)
+                .i32_const(arg_name_len as i32)
+                .global_set(get_global(&self.module, "runtime-error-arg-len")?)
+                .i32_const(ErrorMap::NameAlreadyUsed as i32)
+                .call(self.func_by_name("stdlib.runtime-error"));
+        }
```

Comment:
Just so that I'm sure I understand correctly: here, the body of the function is still generated, even if it will directly throw a runtime error at execution?

Reply:
Yes: if we don't we may generate invalid wasm since it would still expect the return type to be the correct one (eg, getting a `InvalidWebAssembly {message: "type mismatch: expected i64, found i32" ...`); to avoid generating the original body, we would need to generate some dummy data with the correct type (after throwing, just to make the wasm compiler happy).

Reply:
What if you just add an `unreachable` right after the call to `stdlib.runtime-error` ?

Reply:
That works, thanks!